### ISSUE TITLE
FR: Notifier Templates and Bindings

### ISSUE THREAD

**Issue by LOZORD** [GoogleCloudPlatform/cloud-build-notifiers#95]
Hello from the Cloud Build team 👋 

We are aware that many Notifiers users would like to have custom templating and config bindings available to make their notifications more useful. Our product team and I are working on a requirements doc for these features that we plan on releasing here to open up the coding of this feature to the public.

I'll respond here with the doc once we have it in a good state.

**Comment by iamyojimbo** [comment]
Hey @LOZORD, any movement on this?

**Comment by LOZORD** [comment]
Hi Savvas, unfortunately I can't give an update as I no longer work on Cloud Build. @Saaarah might have a better idea.

**Comment by jakebiesinger-storyhealth** [comment]
Friendly ping on this, @Saaarah. 

We're heavy users of cloud build and Slack and our notifications look like garbage and are really hard to tell apart:

![image](https://user-images.githubusercontent.com/66106280/123912457-e85d8500-d931-11eb-91d0-f70d046ed601.png)

We use cloud build + Slack notifications for:
- automated nightly deployments to test environments
- automated daily e2e tests (dev, staging)
- automated code/vulnerability scanners (on several branches)
- to notify about manual deployments as they happen / are successful (dev / staging / prod / demo / etc)
- pull requests (which don't notify)

If standardizing this is too much to take on now, WDYT about just letting the various projects do their thing? We already have implementations ready but which are bitrotting like https://github.com/GoogleCloudPlatform/cloud-build-notifiers/pull/91

Or how about we just start with something simple like "subject/title + body in plaintext" for the payload?

**Comment by cristian-moreno-ruiz** [comment]
Any plans on moving this forward? We are using Cloud Build and have 10+ microservices, in different environments (staging, prod, etc). It would be really helpful to be aware of which of the services was deployed.

Thanks!

**Comment by wyardley** [comment]
Would be very useful to have this.

Also, I'd also suggest that it would be nice to have a simpler, container-based notifications that could either be directly part of Cloudbuild without creating any other resources, or a container based notification setup be embedded directly into workflows (similar to the Slack orb for CircleCI - https://circleci.com/developer/orbs/orb/circleci/slack).

Currently, the number of resources to setup and manage to get this working is quite a hassle for something that IMO should just be a builtin feature. The fact that the actual notifications themselves are not templatable or useful makes it even worse.

**Comment by omrta-dev** [comment]
Is there any progress on this, as others have stated the current notifications are kind of useless as they still require someone to view the logs to find out what service broken. In cases where multiple builds are failing, some of which are more/less important than others it would be very helpful to have some basic information added to the build notifications. Is there any work pending on the cloud build message that gets sent or work implementing the templating for the message?

**Comment by mikehardenize** [comment]
I'm amazed this was released without this functionality. Completely useless. Now I need to back it all out again.

**Comment by nbellamy-paper** [comment]
I see this commit 8 days ago https://github.com/GoogleCloudPlatform/cloud-build-notifiers/commit/b0668daa20c1f653e240545f8a289f12b027fab1

Does that mean that work is in progress for this?

The slack notifications are almost useless with the build ID instead of the friendly name.

**Comment by jfragoulis** [comment]
https://github.com/GoogleCloudPlatform/cloud-build-notifiers/issues/32#issuecomment-1152328261

**Comment by stevejalim** [comment]
I'm seeing more recent commits (early October 2022) and the current state of the readme has this https://github.com/GoogleCloudPlatform/cloud-build-notifiers/commit/49a019d09335cd6190e61c1f05063470ea6c29f6 which is also very promising. Fingers crossed templating is on the way to GA soon!
